src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -89,6 +90,55 @@ accountFromString(AccountID& result, std::string const& strIdent, bool bStrict)
         return Json::objectValue;
 }
 
+std::uint64_t
+getStartHint(std::shared_ptr<SLE const> const& sle, AccountID const& accountID)
+{
+    if (sle->getType() == ltRIPPLE_STATE)
+    {
+        if (sle->getFieldAmount(sfLowLimit).getIssuer() == accountID)
+            return sle->getFieldU64(sfLowNode);
+        else if (sle->getFieldAmount(sfHighLimit).getIssuer() == accountID)
+            return sle->getFieldU64(sfHighNode);
+    }
+
+    return sle->getFieldU64(sfOwnerNode);
+}
+
+bool
+isOwnedByAccount(
+    ReadView const& ledger,
+    std::shared_ptr<SLE const> const& sle,
+    AccountID const& accountID)
+{
+    if (sle->getType() == ltRIPPLE_STATE)
+    {
+        return (sle->getFieldAmount(sfLowLimit).getIssuer() == accountID) ||
+            (sle->getFieldAmount(sfHighLimit).getIssuer() == accountID);
+    }
+    else if (sle->isFieldPresent(sfAccount))
+    {
+        return sle->getAccountID(sfAccount) == accountID;
+    }
+    else if (sle->getType() == ltSIGNER_LIST)
+    {
+        auto hint = sle->getFieldU64(sfOwnerNode);
+        auto const rootIndex = keylet::ownerDir(accountID);
+        auto const pageIndex = keylet::page(rootIndex, hint);
+
+        auto directory = ledger.read({ltDIR_NODE, pageIndex.key});
+
+        if (!directory)
+            return false;
+
+        auto const& hashes = directory->getFieldV256(sfIndexes);
+        auto iter = std::find(hashes.begin(), hashes.end(), sle->key());
+
+        return iter != hashes.end();
```

Comment:
It looks like the way it is currently coded will work, but I think it's the hard way to do it.  I _think_ all you need to check signer lists is this:
```
        Keylet const accountSignerList = keylet::signers(accountID);
        return sle->key() == accountSignerList.key;
```
The `ltSIGNER_LIST` `else if` is not exercised by any unit tests as far as I can tell.  So I can't easily prove that either of these codings is correct.  If there were a unit test I'd verify whether my proposed change works.  As it is we're driving through fog...

Reply:
I went ahead and added some unit tests for his branch.